web/src/main/java/org/springframework/security/web/header/writers/ClearSiteDataHeaderWriter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.header.writers;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.security.web.header.HeaderWriter;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.util.Assert;
+
+/**
+ * Provides support for <a href="https://w3c.github.io/webappsec-clear-site-data/">Clear
+ * Site Data</a>.
+ *
+ * <p>
+ * Developers may instruct a user agent to clear various types of relevant data by delivering
+ * a Clear-Site-Data HTTP response header in response to a request.
+ * <p>
+ *
+ * <p>
+ * Due to <a href="https://w3c.github.io/webappsec-clear-site-data/#incomplete">Incomplete Clearing</a>
+ * section the header is only applied if the request is secure.
+ * </p>
+ *
+ * @author Rafiullah Hamedy
+ * @since 5.2.0
+ */
+public class ClearSiteDataHeaderWriter implements HeaderWriter {
+
+	private static final String CLEAR_SITE_DATA_HEADER = "Clear-Site-Data";
+
+	private final Log logger = LogFactory.getLog(getClass());
```

Comment:
Thank you for looking so carefully at the pattern in other classes and maintaining consistency. Let's use `slf4j`, though, so that we are "coding to the interface".

Reply:
I could not address this comment! There might be some weird dependency issue 😞 

When replacing `Log logger` with `slf4j` alternative, there is no problem in Eclipse and the tests run but, any `gradle` commands fails with `error: package org.slf4j does not exist` (also tried deleting `gradle cache`). 
I looked at the `spring-security-web.gradle` and there is no mention of `slf4j` and also looking at the `git history` of `spring-security-web.gradle` the `slf4jDependencies` was there but, then it was removed. 

Simply adding the dependency back does not make a difference because, the gradle command keeps failing. I tried (from inside `web` directory)
```
../gradlew clean build 
../gradlew clean build test 
../gradlew build 
``` 
and they all complain about the `package org.slf4j`. 

I did a quick search and turns out no other class in `web` is using `slf4j` and they all use the `apache commons` 🤔 

What do you propose I should do?

Reply:
On second thought, yes, let's address the logging strategy of the header writers as a separate unit of work. Sorry to create some heartburn for you.

Feel free to leave it using apache commons for this PR.

Reply:
Sounds good! Please let me know if there is anything outstanding in this PR (It's currently using the Apache Commons).